sqlx-core/src/error/database.rs

```diff
@@ -9,4 +9,65 @@ use std::error::Error as StdError;
 pub trait DatabaseError: 'static + StdError + Send + Sync {
     /// Returns the primary, human-readable error message.
     fn message(&self) -> &str;
+
+    #[doc(hidden)]
+    fn as_error(&self) -> &(dyn StdError + Send + Sync + 'static);
+
+    #[doc(hidden)]
+    fn as_error_mut(&mut self) -> &mut (dyn StdError + Send + Sync + 'static);
```

Comment:
`as_error_mut` is unused, do we need this one?

Reply:
I think the macros used it, might as well leave it in, we can remove it later easily enough since it's not part of the stable API.